src/profiler.cpp

```diff
@@ -120,13 +120,13 @@ void Profiler::addRuntimeStub(const void* address, int length, const char* name)
     CodeHeap::updateBounds(address, (const char*)address + length);
 }
 
-void Profiler::onThreadStart(jvmtiEnv* jvmti, JNIEnv* jni, jthread thread) {
+void Profiler::onThreadStart(jvmtiEnv* jvmti, JNIEnv* jni, jthread thread, int cpu) {
```

Comment:
The signature of this method matches JVM TI [ThreadStart](https://docs.oracle.com/en/java/javase/11/docs/specs/jvmti.html#ThreadStart) event. Therefore, the value of `cpu` is undefined here, which probably results in `EINVAL` when coming to `perf_event_open` syscall.